extensions/amp-sidebar/1.0/sidebar.js

```diff
@@ -175,13 +219,14 @@ function SidebarWithRef(
         sidebarAnimation = null;
         backdropAnimation = null;
         setMounted(false);
+        setSide(null);
```

Comment:
Reset the side, so we can calculate a new side on the next animation.  Captures changes to `document.ltr` and potential other changes that may effect side between each animation.

Reply:
I'm not convinced it is good to reset `side` from the performance perspective, as it causes an additional render only to leave us with less information. However, if this is preferred, it looks like `setMounted` and `setSide` are often called at similar times, likely causing double-rendering already - can this logic be deduped at all to do the work of both in a single render? Especially given the `useLayoutEffect` which calls `setSide` relies on `mounted`: https://github.com/ampproject/amphtml/blob/374121f38750e9318b7b69357778ce81e4c300c7/extensions/amp-sidebar/1.0/sidebar.js#L160

Reply:
I don't think this is needed. Once we initialize the side, we should just stick with it.

Reply:
Okay that's fine, we can remove, but the tradeoff becomes that we do not capture any changes to `document.ltr` that happen after the first time sidebar is opened.

@caroqliu Thanks.  In this case `setMounted` and `setSide` here will trigger one additional render (but not an extra one) so we don't lose anything in terms of performance.  This render will always happen (in order to delete the sidebar elements after animation).

However, we would be calculating the `side` state variable again next time sidebar is opened (that's the point of this line of code).  And that calculation is what causes the second render on sidebar open.  Slightly less performant, but we do get to capture any changes to `document.dir` which we talked about earlier in the review.

Personally I do not have strong opinions either way so I'm fine with removing this line.  Especially since changes to `document.dir` seem pretty unlikely.